### ISSUE TITLE
"conflicting annotations" for declaration @Nonnull on a type-variable usage with type-parameter declaration `extends @Nullable Object`

### ISSUE THREAD

**Issue by cpovirk** [eisop/checker-framework#204]
```
$ cat A.java 
import org.checkerframework.checker.nullness.qual.Nullable;
import javax.annotation.Nonnull;

class A<T extends @Nullable Object> {
  @Nonnull T t;

  A(@Nonnull T t) {
    this.t = t;
  }
}

$ checker/bin/javac -cp $HOME/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar -processor nullness A.java
A.java:5: error: [type.invalid.conflicting.annos] invalid type: conflicting annotations [@Initialized, @NonNull, @Nullable] in type "@Initialized @NonNull @Nullable Object"
  @Nonnull T t;
             ^
A.java:7: error: [type.invalid.conflicting.annos] invalid type: conflicting annotations [@Initialized, @NonNull, @Nullable] in type "@Initialized @NonNull @Nullable Object"
  A(@Nonnull T t) {
               ^
A.java:8: error: [type.invalid.conflicting.annos] invalid type: conflicting annotations [@Initialized, @NonNull, @Nullable] in type "@Initialized @NonNull @Nullable Object"
    this.t = t;
             ^
3 errors
```

That's with 3.21.3-eisop1. It doesn't reproduce against typetools 3.21.3, so I'm assuming it's an accidental result of the eisop improvements to `extends @Nullable Object`.

This is very minor.

The problem disappears if I use a type-use `@NonNull` annotation instead of a declaration one. So there's an easy workaround.

Additionally, this has affected only one file in my testing so far, and that file shouldn't have been using `extends @Nullable Object` in the first place :)

**Comment by wmdietl** [comment]
Thanks for the test case!
This actually gets fixed by #195, which makes that cleanup more important and more concrete.

**Comment by wmdietl** [comment]
@cpovirk I've fixed and merged #195 and included an expanded version of this test case.
Please do let me know if you're running into any other issues!
